forge-ai/src/main/java/forge/ai/simulation/GameStateEvaluator.java

```diff
@@ -148,7 +161,20 @@ public int evalCard(Game game, Player aiPlayer, Card c) {
         if (c.isCreature()) {
             return eval.evaluateCreature(c);
         } else if (c.isLand()) {
-            return 100;
+            int value = 100;
+            // for each mana color a land generates for free, increase the value by one
+            // for each mana a land can produce, add one hundred.
+            int max_produced = 0;
+            Set<String> colors_produced = new HashSet<>();
+            for (SpellAbility m: c.getManaAbilities()) {
+                max_produced = max(max_produced, m.amountOfManaGenerated(true));
+                for (AbilityManaPart mp : m.getAllManaParts()) {
+                    colors_produced.addAll(Arrays.asList(mp.mana(m).split(" ")));
+                }
+            }
+            value = 100 * max_produced;
+            value += colors_produced.size();
```

Comment:
This is a good start. Maybe the next iteration can balance based on what mana you already have available to you. e.g. Playing an Island when you already have 4 Islands out is probably less good than playing a Swamp.

Reply:
That's what the todos in the section above are about. They'd need to be a bit more in-depth though, and would need to apply to the player's state as a whole, not just to each individual land.

Reply:
That's what the todos in the section above are about. They'd need to be a bit more in-depth though, and would need to apply to the player's state as a whole, not just to each individual land.